__tests__/app.test.js

```diff
@@ -32,3 +33,40 @@ describe('/api/topics', () => {
     })
 
 })
+
+
+describe('GET /api', () => {
+    test('responds with an object describing all the available endpoints', () => {
+        return request(app).get('/api').expect(200).then(({ body }) => {
+            const endPoints = body.endpoints;
+            for (const key in endPoints) {
+                expect(endPoints[key].hasOwnProperty('description')).toBe(true);
+            }
+            expect(typeof endPoints).toBe('object');
+        })
+    })
+})
+
+describe('GET /api/articles/:article_id', () => {
+    test('200: responds with an article object with the correct properties when given a valid id that exists', () => {
+        return request(app).get('/api/articles/4').expect(200).then(({ body }) => {
+            const { articles: article } = body;
+            const articleProperties = ['author', 'title', 'article_id', 'body', 'topic', 'created_at', 'votes', 'article_img_url']
+            articleProperties.forEach((articleProperty) => {
+                expect(article[0].hasOwnProperty(articleProperty));
+            })
+
```

Comment:
I see what you are doing in this test, however, it would be a brittle test if we can also test the types of the values in the object. Have a look at ```toMatchObject``` to help you with this

Also, the test suggests that the response value is an array. As it is a single article, an array isn't needed and the object would suffice 🙂

Reply:
I've had a look at toMatchObject() but can't see why to use this instead of toEqual()?
If the object should have all of the properties then shouldn't i test for that rather than a subset?
I've done a refactor of this to test the typeof on each property but not sure that's what you meant..